n_hooks.c

```diff
@@ -821,16 +831,57 @@ bool NoteHardReset()
 /*!
   @brief  Perform a JSON request to the Notecard using the currently-set
   platform hook.
-  @param   json the JSON request.
-  @param   jsonResponse (out) A buffer with the JSON response.
+  @param   request the JSON request.
+  @param   response (out) A buffer with the JSON response.
   @returns NULL if successful, or an error string if the transaction failed
   or the hook has not been set.
 */
 /**************************************************************************/
-const char *NoteJSONTransaction(char *json, char **jsonResponse)
+const char *NoteJSONTransaction(char *request, char **response)
 {
     if (notecardTransaction == NULL || hookActiveInterface == interfaceNone) {
         return "i2c or serial interface must be selected";
     }
-    return notecardTransaction(json, jsonResponse);
+    return notecardTransaction(request, response);
+}
+
+/**************************************************************************/
+/*!
+  @brief  Receive bytes over from the Notecard using the currently-set
+  platform hook.
+  @param   buffer A buffer to receive bytes into.
+  @param   size (in/out)
+            - (in) The size of the buffer in bytes.
+            - (out) The length of the received data in bytes.
+  @param   delay Respect delay standard transmission delays.
+  @param   timeoutMs The maximum amount of time, in milliseconds, to wait for
+            serial data to arrive. Passing zero (0) disables the timeout.
+  @param   overflow (out) The contents did not fit inside the provided buffer.
+  @returns  A c-string with an error, or `NULL` if no error ocurred.
+*/
+/**************************************************************************/
+const char *NoteRawReceive(uint8_t *buffer, size_t *size, bool delay, size_t timeoutMs, size_t *overflow)
+{
+    if (notecardRawReceive == NULL || hookActiveInterface == interfaceNone) {
```

Comment:
I don't quite understand why we need to check hookActiveInterface here and in NoteRawTransmit. Isn't it enough to check if the hook we're calling is NULL?

Reply:
This is pattern matching on existing API.

I believe it prevents the problem from it "accidentally working" if you switched away from a given interface.